thunx-spring/src/main/java/eu/xenit/contentcloud/thunx/spring/data/rest/AbacRepositoryInvokerAdapter.java

```diff
@@ -139,8 +141,59 @@ private String toAlias(Class<?> subjectType) {
     }
 
     private BooleanExpression idExpr(Object id, PathBuilder entityPath) {
-        Field idField = BeanUtils.findFieldWithAnnotation(domainType, Id.class);
+        Field idField = DomainObjectUtils.getIdField(domainType);
         PathBuilder idPath = entityPath.get(idField.getName(), id.getClass());
         return idPath.eq(Expressions.constant(id));
     }
+
+
+    static class DomainObjectUtils {
+
+        private static final boolean JAVAX_PERSISTENCE_PRESENT = ClassUtils.isPresent(
+                "javax.persistence.Id", DomainObjectUtils.class.getClassLoader());
+
+        static final Field getIdField(Class<?> domainClass) {
+
+            Optional<Field> javaxPersistenceId = JAVAX_PERSISTENCE_PRESENT
+                    ? DomainObjectUtils.findFieldWithAnnotation(domainClass, javax.persistence.Id.class)
+                    : Optional.empty();
+
+            return javaxPersistenceId
+                    .or(() -> DomainObjectUtils.findFieldWithAnnotation(domainClass, org.springframework.data.annotation.Id.class))
+                    .orElse(null);
+        }
+
+        private static Optional<Field> findFieldWithAnnotation(Class<?> domainObjClass,
+                                                                 Class<? extends Annotation> annotationClass)
+                throws SecurityException, BeansException {
+
+            // First look for the annotation on the accessor methods
+            BeanWrapper wrapper = new BeanWrapperImpl(domainObjClass);
```

Comment:
According to [the docs](https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/beans/BeanWrapperImpl.html), you should instead get the wrapper via 
```PropertyAccessorFactory.forBeanPropertyAccess(domainObjClass);```

I guess the result will be the same, but more future-proof.

Reply:
No, that is incorrect.

`PropertyAccessorFactory.forBeanPropertyAccess(object)` accepts an object instance, not the `Class<?>` - they call a different overloaded constructor of `BeanWrapperImpl`